int32 object
int32 NAO
int32 hand
---
